From e9214cfb2a247a78a07d2032956cde97e6d19e4a Mon Sep 17 00:00:00 2001
From: lifeng68 <lifeng68@huawei.com>
Date: Tue, 17 Nov 2020 18:37:56 +0800
Subject: [PATCH 11/11] rootfs: add make private for root.path parent

Signed-off-by: lifeng68 <lifeng68@huawei.com>
---
 src/lxc/conf.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/lxc/conf.c b/src/lxc/conf.c
index 3d8713954..ce5bab9c5 100644
--- a/src/lxc/conf.c
+++ b/src/lxc/conf.c
@@ -1434,6 +1434,10 @@ static int lxc_mount_rootfs(struct lxc_conf *conf)
 	* not propagate in other namespaces. Also it will help with kernel
 	* check pass in pivot_root. (IS_SHARED(new_mnt->mnt_parent))
 	*/
+	ret = rootfs_parent_mount_private(conf->rootfs.path);
+	if (ret != 0) {
+		return log_error(-1, "Failed to make parent of rootfs %s to private.", conf->rootfs.path);
+	}
 	ret = rootfs_parent_mount_private(conf->rootfs.mount);
 	if (ret != 0) {
 		return log_error(-1, "Failed to make parent of rootfs %s to private.", conf->rootfs.mount);
-- 
2.25.1

