From 5a3bec3f80d59dfcc76e16cbab957f4072601816 Mon Sep 17 00:00:00 2001
From: wujing <wujing50@huawei.com>
Date: Tue, 5 Jan 2021 16:53:40 +0800
Subject: [PATCH] fix compilation errors without libcap

Signed-off-by: wujing <wujing50@huawei.com>
---
 src/lxc/conf.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/lxc/conf.c b/src/lxc/conf.c
index c3610ae3..19e193dd 100644
--- a/src/lxc/conf.c
+++ b/src/lxc/conf.c
@@ -5301,11 +5301,12 @@ int lxc_drop_caps(struct lxc_conf *conf)
 		goto out;
 	}
 
-#endif
-
 out:
 	free(caplist);
 	return ret;
+#else
+	return 0;
+#endif
 }
 #endif
 
-- 
2.27.0

