From 21aba903a2e0d744eb54a7737bc11585c2be9aa3 Mon Sep 17 00:00:00 2001
From: haozi007 <liuhao27@huawei.com>
Date: Fri, 17 Sep 2021 08:18:14 +0100
Subject: [PATCH] just use origin loop if do not have io

Signed-off-by: haozi007 <liuhao27@huawei.com>
---
 src/lxc/start.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lxc/start.c b/src/lxc/start.c
index e6e2170..52ea561 100644
--- a/src/lxc/start.c
+++ b/src/lxc/start.c
@@ -590,7 +590,7 @@ int lxc_poll(const char *name, struct lxc_handler *handler)
 	}
 #endif
 
-	ret = isulad_safe_mainloop(&descr, -1);
+	ret = lxc_mainloop(&descr, -1);
 	close_prot_errno_disarm(descr.epfd);
 	if (ret < 0 || !handler->init_died)
 		goto out_mainloop_console;
-- 
2.20.1

