From ecb05bff284f2f5364bcfd2196e7cb5e1d4fa512 Mon Sep 17 00:00:00 2001
From: zhangxiaoyu <zhangxiaoyu58@huawei.com>
Date: Fri, 17 Sep 2021 19:45:30 +0800
Subject: [PATCH 2/2] fix lsm_se_mount_context memory leak

Signed-off-by: zhangxiaoyu <zhangxiaoyu58@huawei.com>
---
 src/lxc/conf.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/lxc/conf.c b/src/lxc/conf.c
index 8b1d2d43d..ce550e264 100644
--- a/src/lxc/conf.c
+++ b/src/lxc/conf.c
@@ -5718,6 +5718,7 @@ void lxc_conf_free(struct lxc_conf *conf)
 	if (conf->ocihooks) {
 		free_oci_runtime_spec_hooks(conf->ocihooks);
 	}
+	free(conf->lsm_se_mount_context);
 #endif
 	free(conf);
 }
-- 
2.25.1

