From ff2e749f043f49a4a10d849a56de258a8bfebb56 Mon Sep 17 00:00:00 2001
From: tanyifeng <tanyifeng1@huawei.com>
Date: Wed, 16 Jan 2019 20:53:25 +0800
Subject: [PATCH 035/131] lxc: fixup builds with newer glibc

Signed-off-by: LiFeng <lifeng68@huawei.com>
---
 src/lxc/utils.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/lxc/utils.c b/src/lxc/utils.c
index 120a13d7..c8fb9931 100644
--- a/src/lxc/utils.c
+++ b/src/lxc/utils.c
@@ -44,6 +44,7 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <sys/wait.h>
+#include <sys/sysmacros.h>
 #include <unistd.h>
 #include <pwd.h>
 
-- 
2.23.0

