From 06a4fd31155821e4425a94ed0ff0d2353d5d940d Mon Sep 17 00:00:00 2001
From: wujing <wujing50@huawei.com>
Date: Wed, 15 Apr 2020 23:28:36 -0400
Subject: [PATCH 38/49] Ignore errors when loading rules fail

Signed-off-by: wujing <wujing50@huawei.com>
---
 src/lxc/seccomp.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/lxc/seccomp.c b/src/lxc/seccomp.c
index a75adb7..b7c4de3 100644
--- a/src/lxc/seccomp.c
+++ b/src/lxc/seccomp.c
@@ -586,11 +586,12 @@ bool do_resolve_add_rule(uint32_t arch, char *line, scmp_filter_ctx ctx,
 #ifdef HAVE_ISULAD
 		DEBUG("Failed loading rule for %s (nr %d action %d (%s))",
 		      line, nr, rule->action, get_action_name(rule->action));
+		return true;
 #else
 		SYSERROR("Failed loading rule for %s (nr %d action %d (%s))",
 		         line, nr, rule->action, get_action_name(rule->action));
-#endif
 		return false;
+#endif
 	}
 
 	return true;
-- 
1.8.3.1

