From 5d0b2b47921f1e2c5120dabb9b8916b2abce023b Mon Sep 17 00:00:00 2001
From: LiFeng <lifeng68@huawei.com>
Date: Wed, 16 Jan 2019 05:53:36 -0500
Subject: [PATCH 041/138] return -1 when _lxc_start fails

Signed-off-by: LiFeng <lifeng68@huawei.com>
---
 src/lxc/start.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/lxc/start.c b/src/lxc/start.c
index 3e6854f..7bbcb00 100644
--- a/src/lxc/start.c
+++ b/src/lxc/start.c
@@ -2175,6 +2175,7 @@ int __lxc_start(const char *name, struct lxc_handler *handler,
 
 	if (!handler->init_died && handler->pid > 0) {
 		ERROR("Child process is not killed");
+		ret = -1;
 		goto out_abort;
 	}
 
-- 
1.8.3.1

