From 7fa433ba6e534dcf839845e11771a50c2cc44ff4 Mon Sep 17 00:00:00 2001
From: zhangxiaoyu <zhangxiaoyu58@huawei.com>
Date: Wed, 31 Aug 2022 17:30:27 +0800
Subject: [PATCH] return fail if no args or no rootfs path found

Signed-off-by: zhangxiaoyu <zhangxiaoyu58@huawei.com>
---
 src/lxc/lxccontainer.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/src/lxc/lxccontainer.c b/src/lxc/lxccontainer.c
index 69706a5..dd1dbf2 100644
--- a/src/lxc/lxccontainer.c
+++ b/src/lxc/lxccontainer.c
@@ -1062,6 +1062,20 @@ static bool do_lxcapi_start(struct lxc_container *c, int useinit, char * const a
 		argv = init_cmd = use_init_args(conf->init_argv, conf->init_argc);
 	}
 
+	// do not allow using default rootfs path when isulad
+	if (conf->rootfs.mount == NULL) {
+		ERROR("Empty rootfs path detected");
+		lxc_free_handler(handler);
+		return false;
+	}
+
+	// do not allow using default args when isulad
+	if (!argv) {
+		ERROR("Empty args detected");
+		lxc_free_handler(handler);
+		return false;
+	}
+
 	if (c->image_type_oci) {
 		handler->image_type_oci = true;
 	}
-- 
2.25.1

