From d15c6e2871202d8d8b928a4b4e11537997c24fc1 Mon Sep 17 00:00:00 2001
From: LiFeng <lifeng68@huawei.com>
Date: Thu, 16 Apr 2020 21:39:19 +0800
Subject: [PATCH] clean: add init fd in lxc_init_clean_handler

Signed-off-by: LiFeng <lifeng68@huawei.com>
---
 src/lxc/start.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/lxc/start.c b/src/lxc/start.c
index 5d2faee..938ca8e 100644
--- a/src/lxc/start.c
+++ b/src/lxc/start.c
@@ -2913,7 +2913,9 @@ static struct lxc_handler *lxc_init_clean_handler(char *name, char *lxcpath, str
 	handler->lxcpath = lxcpath;
 	handler->pinfd = -1;
 	handler->sigfd = -EBADF;
+	handler->pidfd = -EBADF;
 	handler->init_died = false;
+	handler->monitor_status_fd = -EBADF;
 	handler->pid = pid;
 	handler->state_socket_pair[0] = handler->state_socket_pair[1] = -1;
 	if (handler->conf->reboot == REBOOT_NONE)
-- 
1.8.3.1

