From 45948760921dbaa1c030b6848168b89428ca1434 Mon Sep 17 00:00:00 2001
From: songbuhuang <544824346@qq.com>
Date: Wed, 8 Feb 2023 14:46:44 +0800
Subject: [PATCH] fix lxc write error message

Signed-off-by: songbuhuang <544824346@qq.com>
---
 src/lxc/cgroups/isulad_cgfsng.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lxc/cgroups/isulad_cgfsng.c b/src/lxc/cgroups/isulad_cgfsng.c
index f5cf81c..6ccff63 100644
--- a/src/lxc/cgroups/isulad_cgfsng.c
+++ b/src/lxc/cgroups/isulad_cgfsng.c
@@ -2463,7 +2463,7 @@ retry:
 			goto retry;
 		}
 		lxc_write_error_message(ops->errfd,
-		                        "%s:%d: setting cgroup config for ready process caused \"failed to write %s to %s: %s\".",
+		                        "%s:%d: setting cgroup config for ready process caused failed to write %s to %s: %s",
 		                        __FILE__, __LINE__, value, fullpath, strerror(errno));
 	}
 	free(fullpath);
-- 
2.25.1

