From 8c232b149654907867c12f10d801769f66d3500c Mon Sep 17 00:00:00 2001
From: gaohuatao <gaohuatao@huawei.com>
Date: Wed, 13 May 2020 23:19:25 -0400
Subject: [PATCH] lxc: disable terminal stdout/stderr pipe O_NONBLOCK

Signed-off-by: gaohuatao <gaohuatao@huawei.com>
---
 src/lxc/terminal.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/lxc/terminal.c b/src/lxc/terminal.c
index 3d2960be..7441de79 100644
--- a/src/lxc/terminal.c
+++ b/src/lxc/terminal.c
@@ -1668,12 +1668,12 @@ int lxc_terminal_create(struct lxc_terminal *terminal)
 			goto err;
 		}
 		/* for stdout */
-		if (pipe2(terminal->pipes[1], O_NONBLOCK | O_CLOEXEC)) {
+		if (pipe2(terminal->pipes[1], O_CLOEXEC)) {
 			ERROR("Failed to create stdout pipe");
 			goto err;
 		}
 		/* for stderr */
-		if (pipe2(terminal->pipes[2], O_NONBLOCK | O_CLOEXEC)) {
+		if (pipe2(terminal->pipes[2], O_CLOEXEC)) {
 			ERROR("Failed to create stderr pipe");
 			goto err;
 		}
-- 
2.19.1

