From e57686a8b94905829fc295981bc8d0a1253f186b Mon Sep 17 00:00:00 2001
From: wangkang <wangkang101@huawei.com>
Date: Fri, 22 May 2020 10:00:00 +0800
Subject: [PATCH] lxc: set normal ret value when populate exist device
 successfully

---
 src/lxc/conf.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/lxc/conf.c b/src/lxc/conf.c
index f46255a4..6890da12 100644
--- a/src/lxc/conf.c
+++ b/src/lxc/conf.c
@@ -4108,6 +4108,7 @@ static int setup_populate_devs(const struct lxc_rootfs *rootfs, struct lxc_list
 			ret = -1;
 			goto reset_umask;
 		}
+		ret = 0;
 	}
 
 reset_umask:
-- 
2.18.2

