From 05ab7596df7f6c4b10401d4a9006f391176d3898 Mon Sep 17 00:00:00 2001
From: lifeng68 <lifeng68@huawei.com>
Date: Wed, 27 May 2020 17:13:25 +0800
Subject: [PATCH] lxc: config default cgroup pattern to lxc%n

Signed-off-by: lifeng68 <lifeng68@huawei.com>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 438d2923..3e86431d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -496,7 +496,7 @@ AC_ARG_WITH([rootfs-path],
 # cgroup pattern specification
 AC_ARG_WITH([cgroup-pattern],
 	[AS_HELP_STRING([--with-cgroup-pattern=pattern], [pattern for container cgroups])],
-	[with_cgroup_pattern=$withval], [with_cgroup_pattern=['']])
+	[with_cgroup_pattern=$withval], [with_cgroup_pattern=['lxc/%n']])
 
 # The path for the apparmor_parser's cache for generated apparmor profiles
 AC_ARG_WITH([apparmor-cache-dir],
-- 
2.20.1

