From 9d7bdd43ca5710e625f02b2dcc7b178071f4ae2b Mon Sep 17 00:00:00 2001
From: maoweiyong <maoweiyong@huawei.com>
Date: Thu, 25 Apr 2019 21:21:12 +0800
Subject: [PATCH 090/140] lxc: remove unused variable

Signed-off-by: maoweiyong <maoweiyong@huawei.com>
---
 src/lxc/start.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/lxc/start.c b/src/lxc/start.c
index 2b0d43e..3657d4e 100644
--- a/src/lxc/start.c
+++ b/src/lxc/start.c
@@ -2824,8 +2824,6 @@ int do_lxcapi_get_pids(char *name, char *lxcpath, struct lxc_conf *conf, pid_t *
 {
 	int ret = 0;
 	struct lxc_handler *handler = NULL;
-	int retry_count = 0;
-	int max_retry = 10;
 	struct cgroup_ops *cg_ops = NULL;
 
 	handler = lxc_init_pids_handler(name, lxcpath, conf);
-- 
1.8.3.1

