From 37d15f82b25e21c466b6632d8e56b50d60cb0444 Mon Sep 17 00:00:00 2001
From: LiuHao <liuhao27@huawei.com>
Date: Wed, 4 Sep 2019 18:01:39 +0800
Subject: [PATCH 119/131] remove unuse binary

Signed-off-by: LiuHao <liuhao27@huawei.com>
---
 hooks/Makefile.am | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/hooks/Makefile.am b/hooks/Makefile.am
index 5ae73d72..f55c4a90 100644
--- a/hooks/Makefile.am
+++ b/hooks/Makefile.am
@@ -10,15 +10,4 @@ hooks_SCRIPTS = \
 	squid-deb-proxy-client \
 	nvidia
 
-binhooks_PROGRAMS = \
-	unmount-namespace
-
-unmount_namespace_SOURCES = \
-	unmount-namespace.c
-
-if IS_BIONIC
-unmount_namespace_SOURCES += \
-	../src/include/lxcmntent.c ../src/include/lxcmntent.h
-endif
-
 EXTRA_DIST=$(hooks_SCRIPTS)
-- 
2.23.0

