From 54ca113d0c93b819ed20edf83ef85dccd584c87a Mon Sep 17 00:00:00 2001
From: wujing <wujing50@huawei.com>
Date: Tue, 17 Dec 2019 21:24:24 +0800
Subject: [PATCH 130/138] fix snprintf create abstract socket name bug

Signed-off-by: wujing <wujing50@huawei.com>
---
 src/lxc/commands_utils.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lxc/commands_utils.c b/src/lxc/commands_utils.c
index 56ecce7..f48f118 100644
--- a/src/lxc/commands_utils.c
+++ b/src/lxc/commands_utils.c
@@ -144,7 +144,7 @@ int lxc_make_abstract_socket_name(char *path, size_t pathlen,
 	}
 
 	ret = snprintf(offset, len, "%s/%s/%s", lxcpath, name, suffix);
-	if (ret < 0 || ret >= len) {
+	if (ret < 0) {
 		ERROR("Failed to create abstract socket name");
 		return -1;
 	}
-- 
1.8.3.1

